Add SPI to request that a WKWebView ignores mouse move events

cdumez opened the pull request:
#### 4bdd79779dacc345a52b134e09c2a66d5b4a8310
<pre>
Add SPI to request that a WKWebView ignores mouse move events
<a href="https://bugs.webkit.org/show_bug.cgi?id=243231">https://bugs.webkit.org/show_bug.cgi?id=243231</a>
&lt;rdar://95823990&gt;

Reviewed by Tim Horton.

Add SPI to request that a WKWebView ignores mouse move events. Before r290743,
the client app could override [WKWebView mouseMoved:] and drop events. However,
this no longer works since r290743 since WKWebView now gets its mouse move
events via another mechanism.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _ignoresMouseMoveEvents]):
(-[WKWebView _setIgnoresMouseMoveEvents:]):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.h:
(WebKit::WebViewImpl::setIgnoresMouseMoveEvents):
(WebKit::WebViewImpl::ignoresMouseMoveEvents const):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(-[WKMouseTrackingObserver mouseMoved:]):
(-[WKMouseTrackingObserver mouseEntered:]):
(-[WKMouseTrackingObserver mouseExited:]):

Canonical link: <a href="https://commits.webkit.org/252850@main">https://commits.webkit.org/252850@main</a>
</pre>

hortont424 commented on Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
```diff
@@ -202,19 +202,19 @@ - (instancetype)initWithViewImpl:(WebKit::WebViewImpl&)impl
 - (void)mouseMoved:(NSEvent *)event
 {
-    if (_impl)
+    if (_impl && !_impl->ignoresMouseMoveEvents())
```
Not entirely sure why these are here instead of in WebViewImpl::mouseMoved, but I guess either way is fine?

cdumez replied:
Right. I don't mind either way. I'll move it.

cdumez committed:
Add SPI to request that a WKWebView ignores mouse move events
https://bugs.webkit.org/show_bug.cgi?id=243231
<rdar://95823990>

Reviewed by Tim Horton.

Add SPI to request that a WKWebView ignores mouse move events. Before r290743,
the client app could override [WKWebView mouseMoved:] and drop events. However,
this no longer works since r290743 since WKWebView now gets its mouse move
events via another mechanism.

* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _ignoresMouseMoveEvents]):
(-[WKWebView _setIgnoresMouseMoveEvents:]):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.h:
(WebKit::WebViewImpl::setIgnoresMouseMoveEvents):
(WebKit::WebViewImpl::ignoresMouseMoveEvents const):
* Source/WebKit/UIProcess/Cocoa/WebViewImpl.mm:
(-[WKMouseTrackingObserver mouseMoved:]):
(-[WKMouseTrackingObserver mouseEntered:]):
(-[WKMouseTrackingObserver mouseExited:]):

Canonical link: https://commits.webkit.org/252850@main
